integration/integration_tms_upload_test.go

```diff
@@ -0,0 +1,123 @@
+	//Reading TMS credentials from environment
+	tmsServiceKey = os.Getenv("PIPER_tmsServiceKey")
+	if len(tmsServiceKey) == 0 {
+		fmt.Println("Env. variable PIPER_tmsServiceKey is not provided")
+		os.Exit(1)
+	}
+}
+
+func TestTmsUploadIntegrationBinSuccess(t *testing.T) {
+	// success case: run cmd without nodeExtDescriptorMapping
+	readEnv()
+	container := givenThisContainer(t, IntegrationTestDockerExecRunnerBundle{
+		Image:       "devxci/mbtci-java11-node14",
+		User:        "root",
+		TestDir:     []string{"testdata", "TestTmsUploadIntegration/bin_param"},
+		Environment: map[string]string{"PIPER_tmsServiceKey": tmsServiceKey},
+	})
+	defer container.terminate(t)
+
+	err := container.whenRunningPiperCommand("tmsUpload",
+		"--mtaPath=scv_x.mtar",
+		"--nodeName=PIPER-TEST",
+		"--customDescription=Piper integration test",
+		"--mtaVersion=1.0.0",
+		"-v")
+	if err != nil {
+		t.Fatalf("Piper command failed %s", err)
+	}
+	container.assertHasOutput(t, "description: Piper integration test")
+	container.assertHasOutput(t, "tmsUpload - File uploaded successfully")
+	container.assertHasOutput(t, "tmsUpload - Node upload executed successfully")
+	container.assertHasOutput(t, "tmsUpload - SUCCESS")
+}
+
+func TestTmsUploadIntegrationBinNoDescriptionSuccess(t *testing.T) {
+	// success case: run cmd without --nodeExtDescriptorMapping and --customDescription
+	readEnv()
+	container := givenThisContainer(t, IntegrationTestDockerExecRunnerBundle{
+		Image:       "devxci/mbtci-java11-node14",
+		User:        "root",
+		TestDir:     []string{"testdata", "TestTmsUploadIntegration/bin_param"},
+		Environment: map[string]string{"PIPER_tmsServiceKey": tmsServiceKey},
+	})
+	defer container.terminate(t)
+
+	err := container.whenRunningPiperCommand("tmsUpload",
+		"--mtaPath=scv_x.mtar",
+		"--nodeName=PIPER-TEST",
+		"--mtaVersion=1.0.0",
+		"-v")
+	if err != nil {
+		t.Fatalf("Piper command failed %s", err)
+	}
+	container.assertHasOutput(t, "description: Created by Piper")
+	container.assertHasOutput(t, "tmsUpload - File uploaded successfully")
+	container.assertHasOutput(t, "tmsUpload - Node upload executed successfully")
+	container.assertHasOutput(t, "tmsUpload - SUCCESS")
+}
+
+func TestTmsUploadIntegrationBinFailParam(t *testing.T) {
+	// run cmd with nodeExtDescriptorMapping
+	readEnv()
+	container := givenThisContainer(t, IntegrationTestDockerExecRunnerBundle{
+		Image:   "devxci/mbtci-java11-node14",
+		User:    "root",
+		TestDir: []string{"testdata", "TestTmsUploadIntegration/bin_param"},
+	})
+	defer container.terminate(t)
+
+	err := container.whenRunningPiperCommand("tmsUpload",
+		"--mtaPath=scv_x.mtar",
+		"--nodeName=PIPER-TEST",
+		"--customDescription=Piper integration test",
+		"--nodeExtDescriptorMapping={\"PIPER-TEST\":\"scv_x.mtaext\", \"PIPER-PROD\":\"scv_x.mtaext\"}",
+		"--mtaVersion=1.0.0",
+		"-v")
+
+	assert.Error(t, err, "Did expect error")
+	container.assertHasOutput(t, "Error: unknown flag: --nodeExtDescriptorMapping")
```

Comment:
This fails in Piper, right? It does not send a request. So this is not really an integration test, more of a unit test. Is there a way to get an error response from the TMS backend? Maybe by providing a faulty mtar file?

Reply:
added a failing test